Soccar [Client]/src/nl/soccar/ui/fx/controller/MainMenuFXMLController.java

```diff
@@ -85,7 +101,24 @@ public void updateTable() {
             sessionItems.add(new SessionTableItem(s));
         }
         tblSessionList.getItems().addAll(sessionItems);
+    }
+
+    public void joinRoom(TableRow row) {
+        SessionTableItem rowData = (SessionTableItem) row.getItem();
+        System.out.println(rowData.getSession().getRoom().getName());
+        if (rowData.getSession().getRoom().passwordAvailable()) {
+            TextInputDialog dialog = new TextInputDialog("Password");
+            dialog.setTitle("Password locked room");
+            dialog.setHeaderText("This room is locked!");
+            dialog.setContentText("Please enter your password:");
+
+            Optional<String> result = dialog.showAndWait();
+            result.ifPresent(password -> sessionController.setCurrentSession(sessionController.join(rowData.getSession(), password, Soccar.getInstance().getCurrentPlayer())));

```

Comment:
rowData.getSession() wordt hier aangevraagd, wordt in de if-statement ook al aangevraagd. -> lokale variabele.